app/recipes_book/views.py

```diff
@@ -0,0 +1,74 @@
+from . import models
+from django.http import JsonResponse, Http404, HttpResponse
+from django.views import View
+import json
+
+
+class RecipesView(View):
+    def get(self, request, recipe_id=None):
+        if recipe_id:
+            recipe = self._load_by_id(recipe_id)
+            return JsonResponse(recipe)
+        elif request.GET.get("name"):
+            filter_name = request.GET.get("name")
+            recipes = self._load_recipes_by_name(filter_name)
+            return JsonResponse(recipes, safe=False)
+        else:
+            recipes = self._load_all_recipes()
+            return JsonResponse(recipes, safe=False)
+
+    def _load_by_id(self, recipe_id):
+        try:
+            recipe = models.Recipe.objects.prefetch_related('ingredients').get(id=recipe_id)
+            return recipe.to_dict()
+        except models.Recipe.DoesNotExist:
+            raise Http404("Recipe not found")
+
+    def _load_all_recipes(self):
+        recipes = models.Recipe.objects.prefetch_related('ingredients').all()
+        recipes_result = [recipe.to_dict() for recipe in recipes]
+        return recipes_result
+
+    def _load_recipes_by_name(self, name):
+        recipes = models.Recipe.objects.filter(name__icontains=name)
+        recipes_result = [recipe.to_dict() for recipe in recipes]
+        return recipes_result
+
+    def post(self, request):
+        data = json.loads(request.body)
+        recipe = models.Recipe.objects.create(
+            name=data["name"], description=data["description"]
+        )
```

Comment:
Should there be some validation here? Like no empty name or description?

Reply:
Makes sense. I will create a test to cover this too.